api/src/main/java/io/grpc/RequestValidationListener.java

```diff
@@ -0,0 +1,54 @@
+package io.grpc;
+
+import com.google.protobuf.MessageLiteOrBuilder;
+
+public class RequestValidationListener<ReqT, ResT> extends ForwardingServerCallListener.SimpleForwardingServerCallListener<ReqT> {
+
+    private ServerCall<ReqT, ResT> serverCall;
+    private Metadata headers;
+    private RequestValidatorResolver requestValidatorResolver;
+
+    public RequestValidationListener(
+            ServerCall.Listener<ReqT> delegate,
+            ServerCall<ReqT, ResT> serverCall,
+            Metadata headers,
+            RequestValidatorResolver requestValidatorResolver
+    ) {
+        super(delegate);
+        this.serverCall = serverCall;
+        this.headers = headers;
+        this.requestValidatorResolver = requestValidatorResolver;
+    }
+
+    @Override
+    public void onMessage(ReqT message) {
+        MessageLiteOrBuilder convertMessage = (MessageLiteOrBuilder) message;
+        RequestValidator<MessageLiteOrBuilder> validator = requestValidatorResolver.find(convertMessage.getClass().getTypeName());
+
+        if (validator == null) {
+            super.onMessage(message);
```

Comment:
nit: `return` within this block so you don't need an else. (reduces indentation)